lib/index.js

```diff
@@ -908,7 +908,14 @@ exports.format = Util.format;
 
 exports.transform = function (source, transform, options) {
 
-    exports.assert(source === null || source === undefined || typeof source === 'object', 'Invalid source object: must be null, undefined, or an object');
+    exports.assert(source === null || source === undefined || typeof source === 'object' || Array.isArray(source), 'Invalid source object: must be null, undefined, an object, or an array');
+
+    if (Array.isArray(source)){
+        return source.map(function (obj){
```

Comment:
node v4 seems to have a different performance profile for this but worth making sure.